java/ql/src/semmle/code/java/Expr.qll

```diff
@@ -638,25 +638,62 @@ class BooleanLiteral extends Literal, @booleanliteral {
   override string getAPrimaryQlClass() { result = "BooleanLiteral" }
 }
 
-/** An integer literal. For example, `23`. */
+/**
+ * An integer literal. For example, `23`.
+ *
+ * An integer literal can never be negative except when:
+ * - It is written in binary, octal or hexadecimal notation
+ * - It is written in decimal notation, has the value `2147483648` and is preceded
+ *   by a minus; in this case the value of the IntegerLiteral is -2147483648 and
+ *   the preceding minus will *not* be modeled as `MinusExpr`.<br/>
```

Comment:
Using `<br/>` here because formatting the query remove trailing <code>&nbsp;&nbsp;</code> (two whitespaces), and QLDoc uses CommonMark which [should support raw HTML](https://spec.commonmark.org/0.29/#raw-html).

LGTM Query Console seems to support it, but the VSCode extension does not, yet (?).